.github/workflows/pipeline.yml

```
@@ -0,0 +1,51 @@
+
+    - name: Run Unit Tests with Coverage
+      run: dotnet test --configuration Release --no-build \
+        --collect:"XPlat Code Coverage" \
+        --logger trx \
+        --results-directory TestResults
+
+    - name: Publish Application
+      run: dotnet publish --configuration Release --output publish
+
+    - name: Upload Build Artifacts
+      uses: actions/upload-artifact@v3
```

actionlint (1.7.4)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)